lib/src/eol.rs

```diff
@@ -0,0 +1,391 @@
+                    .read(true)
+                    .open(file_path)
+                    .map_err(|source| {
+                        Box::new(EolError {
+                            message: format!("failed to open the file {}", file_path.display()),
+                            source: Some(Box::new(source)),
+                        })
+                    })?;
+                let mut content = vec![];
+                file.take(Self::PROBE_LIMIT)
+                    .read_to_end(&mut content)
+                    .map_err(|source| {
+                        Box::new(EolError {
+                            message: format!(
+                                "failed to read from the file {}",
+                                file_path.display()
+                            ),
+                            source: Some(Box::new(source)),
+                        })
+                    })?;
+                if Stats::from_bytes(&content).is_binary() {
+                    return Ok(TargetEol::PassThrough);
+                }
+                Ok(TargetEol::Lf)
+            }
+        }
+    }
+
+    pub(crate) async fn get_target_eol_for_update(
+        &self,
+        repo_path: &RepoPath,
+        file_id: &FileId,
+    ) -> Result<TargetEol, Box<dyn Error + Send + Sync>> {
+        match self.eol_conversion_settings {
+            EolConversionSettings::None | EolConversionSettings::Input => {
+                Ok(TargetEol::PassThrough)
+            }
+            EolConversionSettings::InputOutput => {
+                let mut content = vec![];
+                self.store
+                    .read_file(repo_path, file_id)
+                    .await
+                    .map_err(|source| {
+                        Box::new(EolError {
+                            message: format!(
+                                "failed to create reader to the {} file from the store",
+                                repo_path.as_internal_file_string()
+                            ),
+                            source: Some(Box::new(source)),
+                        })
+                    })?
+                    .take(Self::PROBE_LIMIT)
+                    .read_to_end(&mut content)
+                    .await
+                    .map_err(|source| {
+                        Box::new(EolError {
+                            message: format!(
+                                "failed to read the {} file from the store",
+                                repo_path.as_internal_file_string()
+                            ),
+                            source: Some(Box::new(source)),
+                        })
+                    })?;
+                if Stats::from_bytes(&content).is_binary() {
+                    return Ok(TargetEol::PassThrough);
+                }
+                Ok(TargetEol::Crlf)
+            }
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+enum EolConversionSettings {
```

Comment:
nit: `Settings` sounds like a struct containing multiple fields. `EolConversion` or `EolConversionMode` might be better.